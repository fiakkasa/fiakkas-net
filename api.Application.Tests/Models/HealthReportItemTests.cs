namespace api.Application.Models.Tests;

public class HealthReportItemTests
{
    [Fact]
    public void Should_Convert_HealthReportItem_To_HealthReportEntry()
    {
        var duration = TimeSpan.FromSeconds(1);
        var healthReportEntry = new HealthReportEntry(
            HealthStatus.Healthy,
            "description",
            duration,
            new Exception("Splash!"),
            new Dictionary<string, object>(),
            ["tag"]
        );

        var result = (HealthReportItem)healthReportEntry;

        result.Status.Should().Be(healthReportEntry.Status);
        result.Duration.Should().Be(healthReportEntry.Duration);
        result.Description.Should().Be(healthReportEntry.Description);
        result.Tags.Should().BeEquivalentTo(healthReportEntry.Tags);

        result.MatchSnapshot();
    }
}
