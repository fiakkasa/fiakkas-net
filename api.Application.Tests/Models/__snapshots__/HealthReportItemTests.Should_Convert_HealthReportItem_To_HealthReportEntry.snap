{
  "Description": "description",
  "Duration": "00:00:01",
  "Status": "Healthy",
  "Tags": [
    "tag"
  ]
}
